Developer_README.md

```diff
@@ -0,0 +1,210 @@
+
+ # Getting started
+ 
+Follow these steps to set up and run the application on your local machine.
+
+### Prerequisites
+Before you begin, make sure you have the following installed:
+
+- [Python](https://www.python.org/downloads/)
+- [Node.js](https://nodejs.org/en/download)
+- [MongoDB](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-windows)
+ 
+ Step 1: 
+  Git Clone the Repository 
+  
+    git clone https://github.com/aditikilledar/burnout-proj3
+
+Step 2:
+   Ensure mongodb is running and execute the below to populate the tables:
+   Navigate to the backend folder using "cd backend".
+
+  Once in the backend, run:
+  
+    python insert_event_data.py
+    python insert_food_data.py
+
+ Step 3:
+   Remain in the backend folder and run the following command to start the server:
+    
+    pip install -r requirements.txt
+    flask run
+
+ Step 4:
+   Navigate to the frontend folder and run the below to start the React app:
+    
+    cd frontend
+    npm install 
+    npm start
+    
+ Step 5:
+    Open the URL in your browser:  
+      http://127.0.0.1:3000
+      
+
+## Third-Party Dependencies
+  
+  Frontend:
+
+  | Package Name        | Version           | License  |
+| ------------- |:-------------:| -----:|
+| material-ui      | 4.12.4 | MIT License |
+| axios      | 1.5.1      |   MIT License |
+
+  Backend:
+
+   | Package Name        | Version           | License  |
+| ------------- |:-------------:| -----:|
+| Flask      | 2.2.5 | BSD 3-Clause |
+| python_dotenv      | 0.21.1      |   BSD License (BSD-3-Clause) |
+| flask_jwt_extended      | 4.5.3      |   MIT License |
+| bcrypt      | 4.0.1      |   Apache License 2.0 |
+| flask_pymongo      | 2.3.0    |   BSD License (BSD) |
+| mongomock      | 4.1.2      |   BSD |
+| flasgger      | 0.9.7.1      |   MIT License |
+| coverage      | 7.2.7      |   Apache Software License (Apache-2.0) |
+
+  
+# What's new?
+
+This version enhanced and built on the previous version, making it 10x more interesting and easier to use!
+Here's what we added in this release.
+
+1. Ability to add custom foods and track their calories.
+2. Fully customizable meal creation.
+3. An intuitive and searchable dropdown.
+4. BMI Calculation for your personal profile.
+5. Intelligent automatic calorie burn calculation.
+6. Enhanced UI.
+7. Randomized Exercise of the Day.
+8. Google Sign-in is now an option!
+9. Better progress tracking on the landing page.
+
+## Bug Fixes
+We fixed many bugs, but here are the most notable ones...
+
+1. Sign up / Sign in issues, especially when an existing user tries to sign up again.
+2. The proxy in the config files were wrong, making it difficult to run the website. Proxy issues were fixed.
+3. Testcases were updated and fixed.
+4. Event were not un-enrollable. Now events can be enrolled and unenrolled.
+5. Exercise of the day was static, which defeats the purpose. Fixed, it is now random.
+6. Linechart on the landing page was incorrect. Fixed.
+
```

Comment:
Can we add another section here for common issues faced during setup.
Ex: npm install --force might be required.
flask not recgnized in the shell/terminal {fix: python -m flask run (python 3.11.6 needed)(to be confirmed)}
etc

Reply:
Sure, I will make those changes on another branch. This PR was to establish a basic structure. Once this is merged, I will start with further changes.

Reply:
Okay. Sounds great.